src/parser.py

```diff
@@ -1,38 +1,114 @@
 from src.tokens import TokenType
-from src.ast_node_expression import Binary, Group, Literal, Unary
 from src.error_handler import ErrorHandler, ParseError
-from src.abstract_syntax_tree import AbstractSyntaxTree
+
+from src.ast_node_expression import Binary, Group, Literal, Unary, Variable, Assign
+
+from src.ast_node_statement import VarStatement, ExpressionStatement, PrintStatement, \
+    BlockStatement, IfStatement, WhileStatement, FunctionStatement, ReturnStatement, \
+    ClassStatement
 from src.parser_constants import EQUALITY_TOKENS, COMPARISON_TOKENS, TERM_TOKENS, FACTOR_TOKENS, \
-    UNARY_TOKENS, LITERAL_TOKENS, IGNORED_TOKENS, GROUP_OPENING, GROUP_CLOSING, \
-    STATEMENT_START_TOKENS, STATEMENT_END_TOKENS
+    UNARY_TOKENS, LITERAL_TOKENS, IGNORED_TOKENS, GROUP_OPENING_TOKENS, GROUP_CLOSING_TOKENS, \
+    STATEMENT_START_TOKENS, STATEMENT_END_TOKENS, IDENTIFIER_TOKENS, EQUALS_TOKENS, \
+    VAR_STATEMENT_TOKENS, \
+    PRINT_STATEMENT_TOKENS
 
 
 class Parser:
     def __init__(self, tokens=None):
-        self.tokens = [token for token in tokens if token not in IGNORED_TOKENS] if tokens else []
+        self.tokens = [token for token in tokens if token.token_type not in IGNORED_TOKENS] if tokens else []
```

Comment:
Why can't we just do self.tokens = tokens here and set the tokens default value to an empty list?